tracing-futures/src/lib.rs

```diff
@@ -287,9 +294,11 @@ impl<T: core::future::Future> core::future::Future for Instrumented<T> {
     type Output = T::Output;
 
     fn poll(self: Pin<&mut Self>, cx: &mut Context<'_>) -> core::task::Poll<Self::Output> {
-        let this = self.project();
+        let mut this = self.project();
         let _enter = this.span.enter();
-        this.inner.poll(cx)
+        // SAFETY: As long as `ManuallyDrop<T>` does not move, `T` wont move.
+        let inner = unsafe { this.inner.as_mut().map_unchecked_mut(|v| &mut **v) };
```

Comment:
nit, not a huge deal: i wondered if it would it be possible to call the `inner_pin_mut` method here, rather than repeating the same `unsafe` code from that method...but we can't do that, because we need to pin-project `self` to access the span, which consumes the `Pin<&mut Self>`...we _might_ want to implement a private method that does this on the generated projection type, since we repeat this unsafe code several times? not a blocker if that's too much extra work.